impl/src/main/kotlin/com/android/declarative/internal/configurators/IncludedBuildPluginCache.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.declarative.internal.configurators
+
+import org.gradle.api.Project
+import java.io.File
+import java.util.Properties
+
+class IncludedBuildPluginInfo(
+    val classesDir: File,
+    val resourcesDir: File,
+)
+
+class IncludedBuildPluginCache(
+    private val project: Project,
+    private val includedBuildDir: File,
+) {
+    private val markerFileLookup = "**/META-INF/gradle-plugins/*.properties"
+
+    private val markerFiles: Set<File> =
+        project.fileTree(includedBuildDir) {
+            it.include(markerFileLookup)
+        }.files
+
+
+    fun getPluginClassPath(): Set<File> =
```

Comment:
As far as I understand this method update cache and resolvePluginById uses this information. Should we rename getPluginClassPath method   to make it more clear?

Reply:
it does not update the cache, the cache is keeping the list of marker files at initialization and getPluginClasspath() uses that information without updating anything.